r/tests/testthat/test-dataset.R

```diff
@@ -548,6 +548,18 @@ test_that("filter() with strings", {
   )
 })
 
+test_that("filter() with arrow compute functions by name", {
+  skip_if_not_available("parquet")
+  ds <- open_dataset(dataset_dir, partitioning = schema(part = uint8()))
+  expect_equivalent(
+    ds %>%
+      select(part, lgl) %>%
+      filter(arrow_is_valid(lgl), arrow_equal(part, 1)) %>%
+      collect(),
+    tibble(part = 1L, lgl = df1$lgl[!is.na(df1$lgl)])
```

Comment:
`is_valid` is equivalent to `!is.na()` (and it's not something you can otherwise access). Is a comment to that effect sufficient, or do you have a different suggestion?

FTR if you search this file for `df1$lgl[!is.na(df1$lgl)]`, you'll see the test I based this on. Happy to use a different test if you think better.

Reply:
I think the equivalence would be clearer if the reference result were created in the same manner as the tested result:
```suggestion
    ds %>%
       select(part, lgl) %>%
       filter(!is.na(lgl), part == 1L) %>%
       collect()
```